packages/commercetools/api-client/src/api/getProduct/index.ts

```diff
@@ -10,33 +8,26 @@ interface ProductData {
   products: ProductQueryResult;
 }
 
-const getProduct = async (search: ProductSearch): Promise<ApolloQueryResult<ProductData>> => {
-  if (search.customQuery) {
-    const { query, variables } = search.customQuery;
-
-    return await apolloClient.query<ProductData>({
-      query: gql`${query}`,
-      variables
-    });
-  }
-
+const getProduct = async (params, resolveQuery = async (query = defaultQuery, variables = {}) => {
   return await apolloClient.query<ProductData>({
-    query: defaultQuery,
+    query: gql`${query}`,
     variables: {
-      where: buildProductWhere(search),
-      skus: search.skus,
-      limit: search.limit,
-      offset: search.offset,
+      where: buildProductWhere(params),
```

Comment:
if the variables are now configurable you have to make sure that if some of them will be removed the code will still work. Try to be more [defensive](https://enterprisecraftsmanship.com/posts/defensive-programming/) here ;)

Reply:
Done.